Pass style with meta completion text

prurph opened the pull request:
Hi there! I love ptpython, love that users can style it, and am hoping you'll consider a small PR to facilitate that.

The `completion-menu.completion` and `completion-menu.meta.completion` appear to be siblings of each other, so there's no way to style the meta menu based on the type of completion. I personally find it much easier to read if the type is colored, but the completion element is always a consistent style, and this enables that. Here is an example:

![CleanShot 2024-07-22 at 23 02 44@2x](https://github.com/user-attachments/assets/769bb78b-f590-4df6-9074-9fb55188a946)

Ignore the `class` there: I modified the types returned as well when I noticed that jedi returns several more options than the three (builtin, param, keyword) that ptpython currently maps to. I have not included that change in this PR. I would be happy to submit another PR for that if it would be welcome.

Thanks for considering!

jonathanslenders commented:
Hi @prurph,

Thanks for this! I totally missed this PR, but this looks very nice! If you're still around and would like to rebase it, I will merge it. (Otherwise, I'll try to do it myself.) I would also not mind if you would add these colors for styling the meta column to the default built-in stylesheet.

prurph commented:
@jonathanslenders thanks for the reply! I've rebased it and added the classes I used to style them to the default stylesheet. I tried to use colors that aligned with what was already being used for these types, and to stick with ansi so it works with dark/light terminal themes, but would be happy to change them. I also added both the container specifier like `completion.param-meta` and the inside characters like `completion.param-meta fuzzymatch.inside.character` specifically to document for folks who might like to add a more elaborate style to their own config.

Let me know if you think it needs any more changes!

jonathanslenders commented:
Perfect! Thanks a lot for this!

jonathanslenders commented:
Actually, it could be that either I'll switch back to RGB for the colors in the completion menu, or use an ANSI color for the background. Contrast for ANSI colors on top of RGB colors is not guaranteed, and the background color for the menu is (I think) defined using RGB.
